src/api-service/__app__/onefuzzlib/proxy.py

```diff
@@ -90,6 +93,23 @@ def init(self) -> None:
                 self.save_proxy_config()
                 self.set_state(VmState.extensions_launch)
         else:
+            nsg = NSG(
+                name=self.region,
+                region=self.region,
+            )
+
+            result = nsg.create()
+            if isinstance(result, Error):
+                self.set_failed(result)
+                return
+
+            result = nsg.set_allowed_sources(["*"])
```

Comment:
This determines important security-relevant defaults, right? We need to extensively document such choices, and ideally also use a named constant.

In this case, don't we want the default behavior to be "deny all incoming IPs"?

Reply:
I think this might have been a placeholder that he was using while waiting for my changes from last week.

Reply:
This is a default for this PR, since the point of this PR is just to add NSG. Applying NSG rules is coming up down the road.
As per @nharpher285 comment